doc/release-notes.rst

```diff
@@ -30,6 +30,12 @@ Documentation
 * Documentation refactorization with examples. :issue:`472` by :user:`Guillaume
   Lemaitre <glemaitre>`.
 
+Maintenance
+...........
+
+* Remove :class:`pydicom.filereader.DicomIter` since it is unused. :issue:`482`
+  by :user:`Guillaume Lemaitre <glemaitre>`.
+
```

Comment:
Right - I'm fine with both mentioning or not mentioning the author, but in a consistent way. As it is never done in the rest of the release notes, I would just stuck to this, but that's just my opinion.